cachito/workers/pkg_manager.py

```diff
@@ -74,13 +75,12 @@ def resolve_gomod_deps(archive_path, copy_cache_to=None):
             if len(parts) == 2:
                 deps.append({'type': 'gomod', 'name': parts[0], 'version': parts[1]})
 
-        if copy_cache_to:
-            # Copy gomod cache to requested location
+        # Add the gomod cache to the bundle the user will later download
+        if request_id is not None:
             cache_path = os.path.join('pkg', 'mod', 'cache', 'download')
             src_cache_path = os.path.join(temp_dir, cache_path)
-            dest_cache_path = os.path.join(
-                worker_config.cachito_shared_dir, copy_cache_to, 'gomod', cache_path)
-            shutil.copytree(src_cache_path, dest_cache_path)
+            dest_cache_path = os.path.join('gomod', cache_path)
+            add_deps_to_bundle_archive(request_id, archive_path, src_cache_path, dest_cache_path)
```

Comment:
It seems a bit odd for `resolve_gomod_deps` to actually create the final bundle. This will be problematic when we add support for other package managers and we want to parallelize dependency resolution.

What if we introduced a new task to finalize the bundle? In that way, each task essentially dumps content in a known location, and the finalize task creates the downloadable archive which simply includes everything from that location.

_This is not a must because currently only gomod is supported._

Reply:
As discussed in IRC. I'll refactor this in a separate PR.